tests/virt_autotest/setup_dns_service.pm

```diff
@@ -18,19 +18,25 @@ use testapi;
 sub run {
     my $self = shift;
 
-    my $wait_script             = "180";
-    my $dns_bash_script_url     = data_url("virt_autotest/setup_dns_service.sh");
-    my $execute_dns_bash_script = "bash <(curl -s $dns_bash_script_url)";
-    script_output($execute_dns_bash_script, $wait_script, type_command => 0, proceed_on_failure => 0);
+    my $wait_script          = "180";
+    my $dns_bash_script_name = "setup_dns_service.sh";
+    my $dns_bash_script_url  = data_url("virt_autotest/$dns_bash_script_name");
+    my $dns_forward_domain   = "testvirt.net";
+    my $dns_reverse_domain   = "123.168.192";
+    my $dns_server_ipaddr    = "192.168.123.1";
+    my $download_bash_script = "curl -s -o ~/$dns_bash_script_name $dns_bash_script_url";
+    script_output($download_bash_script, $wait_script, type_command => 0, proceed_on_failure => 0);
+    my $execute_bash_script = "chmod +x ~/$dns_bash_script_name && ~/$dns_bash_script_name -f $dns_forward_domain -r $dns_reverse_domain -s $dns_server_ipaddr";
+    script_output($execute_bash_script, $wait_script, type_command => 0, proceed_on_failure => 0);
```

Comment:
if this execute_bash_script failed, will the post_fail_hook be triggered? I plan to check this in failure job, but seems the given link can not be opened. Would you please update the link http://10.67.18.219/tests/281?

Reply:
It is valid. You can also refer to http://147.2.217.156/tests/281  or http://147.2.217.156/tests/280   If none works for you, I think that is magic.

Reply:
I can open the two jobs you give. Yes, post_fail_hook is triggered.